lib/manager/npm/post-update/yarn.ts

```diff
@@ -81,21 +82,21 @@ export async function generateLockFile(
     const isYarnDedupeAvailable =
       minYarnVersion && gte(minYarnVersion, '2.2.0');
     const isYarnModeAvailable = minYarnVersion && gte(minYarnVersion, '3.0.0');
-
-    let installYarn = 'npm i -g yarn';
+    const toolConstraint: ToolConstraint = {
+      toolName: 'yarn',
+    };
     if (isYarn1 && minYarnVersion) {
-      installYarn += `@${quote(yarnCompatibility)}`;
+      toolConstraint.constraint = yarnCompatibility;
     }
 
-    const preCommands = [installYarn];
-
     const extraEnv: ExecOptions['extraEnv'] = {
       NPM_CONFIG_CACHE: env.NPM_CONFIG_CACHE,
       npm_config_store: env.npm_config_store,
       CI: 'true',
     };
 
     const commands = [];
+    const preCommands = [];
     let cmdOptions = ''; // should have a leading space
     if (config.skipInstalls !== false) {
       if (isYarn1) {
```

Comment:
We need to check that `yarn` is still installed in the same place and that the `sed` logic below this works

Reply:
yes, the yarn patch can be complicated. my original idea in buildpack was to not install those npm packages to normal global path.

Instead i would like to install to a separate folder like `/usrc/local/yarn/x.y.z` and add the wrapper.
When installing node, i would always create the npm wrapper, so the global npm always points to right version (latest installed).

With seperate folders, we can have multiple versions of those installed at same time and only need to switch the path.

so not really sure how to fix this. maybe add a helper script to buildpack, which does the yarn hack?

Reply:
Another challenge is that there's theoretically the possibility - once we move to binarySource=install that one repo uses slim yarn and the next needs full install. So maybe we should have the concept of "install-tool yarn-slim" and be ready to maintain both patched and non patched versions side by side? Whichever is the latest to be installed then becomes "yarn". Renovate decides which of the two to install each time.

Reply:
That Sounds good. Will be more easy to implement. What about repo local yarn path installs?

Reply:
Those are a temporary clone so ok to repeat it in those